pymeasure/experiment/results.py

```diff
@@ -138,12 +149,54 @@ def format(self, record):
         :type record: dict
         :return: a string
         """
-        return self.delimiter.join('{}'.format(record[x]) for x in self.columns)
+        return self.delimiter.join(
+            '{}'.format(record[x]) for x in self.columns)
 
     def format_header(self):
         return self.delimiter.join(self.columns)
 
 
+class CSVFormatter_Pandas(Results_Formatter):
+    """ Formatter of data results, pandas dataframe or single-line CSV """
+
+    def format(self, record):
+        """Formats a record as csv using pandas built-in ``to_csv`` method.
+        Accepts a pandas dataframe or a dict of values matching
+        the given list of columns.
+
+        :param record: record to format.
+        :type record: pandas.DataFrame or dict
+        :return: str
+        """
+        if isinstance(record, pd.DataFrame):
+            pass
+        elif isinstance(record, dict):
+            # ensure correct order of columns
+            record = (record[c] for c in self.columns)
+            record = pd.DataFrame(record)
```

Comment:
This doesn't seem to work for me, but changing it to the following does work and keep the correct order:
```suggestion
            record = pd.DataFrame([record], columns=self.columns)
```